sabnzbd/utils/ssdp.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/python3 -OO
+# Copyright 2009-2020 The SABnzbd-Team <team@sabnzbd.org>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+"""
+sabnzbd.utils.ssdp - Support for SSDP / Simple Service Discovery Protocol plus XML to appear on Windows
+"""
+import logging
+import time
+import socket
+import uuid
+from threading import Thread
+from typing import Optional
+
+
+class SSDP(Thread):
+    def __init__(self, host, server_name, url, description):
+        self.__host = host  # Note: this is the LAN IP address!
+        self.__server_name = server_name
+        self.__url = url
+        self.__description = description
+        self.__myhostname = socket.gethostname()
+        # uuid stays the same as long as hostname and ip address stay the same:
+        self.__uuidXML = uuid.uuid3(uuid.NAMESPACE_DNS, self.__myhostname + self.__host)
+        self.__uuidSSDP = uuid.uuid3(uuid.NAMESPACE_DNS, self.__host + self.__myhostname)
+
+        self.__stop = False
+        super().__init__()
+
+    def stop(self):
+        logging.info("Stopping SSDP")
+        self.__stop = True
+
+    def run(self):
+        logging.info("Serving SSDP on %s as %s", self.__host, self.__server_name)
+        logging.info("self.__url is %s", self.__url)
+
+        descriptionxmlURL = self.__url + "/description.xml"
+
+        # the standard multicast settings for SSDP:
+        MCAST_GRP = "239.255.255.250"
+        MCAST_PORT = 1900
+        MULTICAST_TTL = 2
+
+        mySSDPbroadcast = f"""NOTIFY * HTTP/1.1
```

Comment:
@Safihre I could move the creation of the SSDP and the XML to the __init__(). That way they are only created once, and the variables used are fixed after __init__ has done its work ...

Reply:
... just changed the code like that.